### ISSUE TITLE
how to block resize when user try to resize less than the default col and width you gave to your component ?

### ISSUE THREAD

**Issue by soludents** [tiberiuzuld/angular-gridster2#457]
Hello I have a case where I need to define a specific mincol and minwidth for each of component and I want my user not to be able to resize less than the mincol and minwidth for each component , is that possible to achieve ?

**Comment by tiberiuzuld** [comment]
Hi @soludents ,
You can set `minItemCols` for an item.
https://tiberiuzuld.github.io/angular-gridster2/items
But there is no min Width see https://github.com/tiberiuzuld/angular-gridster2/issues/228

**Comment by soludents** [comment]
yes I set minItemCols for all items, the probem it applies to all my items.

I also define default size of my items : {cols: 2,hasContent: true,id: "blabla",rows: 3,x: 3,y: 0}

I need to disable resize when it is trying to resize less than the default size I give for each specific item ?

I tried to use callback resizable.stop() but it only occurs onmouseup

**Comment by soludents** [comment]
this is default item size : {cols: 2,hasContent: true,id: "blabla",rows: 3,x: 3,y: 0}

in options I have:
 minItemCols: 2,
 minItemRows: 2

now I don't want the user to be able to resize less than the default size gave them (row 3), right now he will be able to resize up to row 2. However I do want the user to be able to make it bigger if he wants to.

now as i said minitemrow and col applies to all items, I have a lot of items

you see where I am going ? thank you for the answers

**Comment by tiberiuzuld** [comment]
ok, then why not set the `minItemRows` to 3 ?

**Comment by soludents** [comment]
because I have a lot of other items that rely on minItemRows being 2 since I have also item that need to keep  {cols: 2,hasContent: true,id: "blabla2",rows: 2,x: 3,y: 0} , setting it to 3 would override default and cause issues 
my idea is to use resizable stop but no luck with it so far
`resizable: {
        enabled: true,
	stop: (item, gridcomponent, event) => {
          if (item.id = "blablabla") {
            if (gridcomponent.$item.rows < 3) {
              console.log(gridcomponent.$item);
              this.options.resizable.enabled = false;
              //which should block the resize right ?
              return;

            }
          }

        }`

**Comment by soludents** [comment]
how do I reject the stop event ? 

resizable.stop | callback when resizing an item stops. Accepts Promise return to cancel/approve resize

**Comment by soludents** [comment]
gridcomponent.resize.cancelResize()  did the trick.

you can close it, wish your api was a bit more documented :) 

thanks nonetheless your library is very well written.

**Comment by soludents** [comment]
It triggers me however the following error message : 

`core.js:1449 ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'setPushedItems' of undefined
TypeError: Cannot read property 'setPushedItems' of undefined
    at GridsterResizable.push../node_modules/angular-gridster2/fesm5/angular-gridster2.js.GridsterResizable.makeResize (angular-gridster2.js:3778)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:388)
    at Object.onInvoke (core.js:4760)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:387)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:138)
    at zone.js:872
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:4751)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:420)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:188)
    at resolvePromise (zone.js:814)
    at zone.js:877
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:4751)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:420)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:188)
    at drainMicroTaskQueue (zone.js:595)
    at ZoneTask.push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask [as invoke] (zone.js:500)
    at invokeTask (zone.js:1540)
    at HTMLDocument.globalZoneAwareCallback (zone.js:1566)`

**Comment by soludents** [comment]
ok I think I understand why it is happening , might be a bug : 

it is trying to call GridsterResizable.prototype.makeResize  after I triggered this function gridcomponent.resize.cancelResize() inside resizable.stop callback.

i am guessing makeResize should not be called after cancelResize() , makes sense no ?

**Comment by viethien** [comment]
I set minItemCols/minItemRows value for dashboard component item and it worked.

Thanks
